packages/dev/core/src/Physics/v2/IPhysicsEnginePlugin.ts

```diff
@@ -229,16 +237,11 @@ export interface IPhysicsEnginePluginV2 {
     getAngularVelocityToRef(body: PhysicsBody, angVel: Vector3): void;
     getBodyGeometry(body: PhysicsBody): {};
     disposeBody(body: PhysicsBody): void;
-    registerOnBodyCollide(
-        body: PhysicsBody,
-        func: (collider: PhysicsBody, collidedAgainst: PhysicsBody, point: Nullable<Vector3>, distance: number, impulse: number, normal: Nullable<Vector3>) => void
-    ): void;
-    unregisterOnBodyCollide(
-        body: PhysicsBody,
-        func: (collider: PhysicsBody, collidedAgainst: PhysicsBody, point: Nullable<Vector3>, distance: number, impulse: number, normal: Nullable<Vector3>) => void
-    ): void;
     setCollisionCallbackEnabled(body: PhysicsBody, enabled: boolean): void;
     addConstraint(body: PhysicsBody, childBody: PhysicsBody, constraint: PhysicsConstraint): void;
+    getCollisionObservable(
+        body: PhysicsBody
+    ): Observable<{ collider: PhysicsBody; collidedAgainst: PhysicsBody; point: Nullable<Vector3>; distance: number; impulse: number; normal: Nullable<Vector3> }>;

```

Comment:
that's probably not for this PR, but this is a perfect candidate for a getter instead of a get function. the syntax in code will be cleaner when using it.